reference/conanfile/methods/build_id.rst

```diff
@@ -0,0 +1,58 @@
+.. _reference_conanfile_methods_build_id:
+
+build_id()
+==========
+
+The ``build_id()`` method allows to re-use the same build to create different binary packages in the cache,
+potentially saving build time as it can avoid some unnecessary re-builds. It is an optimization method.
+
+In the general case, there is one build folder for each binary package, with the exact same ``package_id`` of the package. However this behavior
+can be changed, there are a couple of scenarios that this might be useful:
+
+- The package build scripts generates several different configurations at once (like both debug and release artifacts) in the same run, without the possibility of building each configuration separately.
+- The package build scripts generate one binary configuration, but different artifacts that can be packaged separately. For example if there are some test executables, you might want to create two packages: one just containing the library for general usage, and another one also containing the tests (for compliance, later reproducibility, debugging, etc).
+
+In the first case, we could write for example:
+
+..  code-block:: python
+
+    settings = "os", "compiler", "arch", "build_type"
+
+    def build_id(self):
+        self.info_build.settings.build_type = "Any"
```

Comment:
should it be `"Any"` or `"ANY"` or `["ANY"]`?
according to https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#any-special-value, the special value is `"ANY"` and it has to be a list :thinking:

Reply:
``Any`` is correct, the upper case ``"ANY"`` is for other thing, definition of possible values. This is a random value.